### PR TITLE
Compute average using each previous value proportionately

### PR DESCRIPTION
**Pull request by tchapeaux**
Previous version used a flawed formula for average that meant that the first values had far less weight than later ones.

Indeed, due to how the .reduce() works, we had the following result:
input [1, 3, 0, 1, 10]
reduce =>
- avg(1, 3) = 2
- avg(2, 0) = 1
- avg(1, 1) = 1
- avg(1, 10) = 5.5

when actually, avg(input) should be equal to 3.

This might actually make sense if we want the latest PRs to have more weight, but in that case we are missing a .sort() function to make sure they are ordered correctly (I can do that PR as well if this is what we prefer)

### CODE REVIEW

File: src/stats.js [hunk 1]
```diff
@@ -27,50 +27,12 @@ const self = (module.exports = {
...
-          fromPreviousStep: averageTimeDiff(
-            a.merged.fromPreviousStep,
-            b.merged.fromPreviousStep
-          ),
-        },
-      };
-    });
+    const result = {}
+
+    ["readyForReview", "firstReview", "firstApprovedReview", "merged"].forEach(column => result[column] = {
```
Comment (manzan46) [inline src/stats.js]: Nice clean up 🙏🏻

**Review by manzan46 — APPROVED** [review]
🙏🏻

File: src/stats.js (hunk 1 above)
Comment (tchapeaux) [reply src/stats.js]: Sure 🙂 For some reason Prettier wasn't working on my end so we might clean it up even more

Comment (manzan46) [reply src/stats.js]: I put the same prettier and eslint as main wooclap, let's merge and I'll iterate on some other stuff

File: src/stats.js [hunk 2]
```diff
@@ -27,12 +27,19 @@ const self = (module.exports = {
         })
     );
 
-    const result = {}
-
-    ["readyForReview", "firstReview", "firstApprovedReview", "merged"].forEach(column => result[column] = {
-      fromPrCreation: averageValues(statsResult.map(sr => sr[column].fromPrCreation)),
-      fromPreviousStep: averageValues(statsResult.map(sr => sr[column].fromPreviousStep))
-    })
+    const result = {}[
```
Comment (tchapeaux) [inline src/stats.js:30]: This doesn't look right.

I think that Prettier merged two statements into one. We should add a `;` after the `{}` to make it clear

```suggestion
    const result = {};
    
    [
```

Comment (manzan46) [reply src/stats.js:30]: 👍 will update in the PR 🙏🏻

**Pull request merged**